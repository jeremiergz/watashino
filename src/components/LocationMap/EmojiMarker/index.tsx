import AstonishedIcon from 'components/svg/emojis/Astonished';
import NerdIcon from 'components/svg/emojis/Nerd';
import UglyIcon from 'components/svg/emojis/Ugly';
import React, { useState } from 'react';

const Emojis = {
  Astonished: AstonishedIcon,
  Nerd: NerdIcon,
  Ugly: UglyIcon,
};

let timeout: number;

const EmojiMarker: React.FC<EmojiMarkerProps> = ({ onClick }) => {
  const [marker, setMarker] = useState(NerdIcon.displayName);
  const Emoji = Emojis[marker];
  const handleMarkerClick = () => {
    if (typeof onClick === 'function') onClick();
    if (marker === NerdIcon.displayName) {
      setMarker(AstonishedIcon.displayName);
    } else if (marker === AstonishedIcon.displayName) {
      setMarker(UglyIcon.displayName);
    }
    clearTimeout(timeout);
    timeout = setTimeout(() => {
      setMarker(NerdIcon.displayName);
    }, 2500);
  };
  return <Emoji cursor="help" onClick={handleMarkerClick} />;
};

EmojiMarker.displayName = 'EmojiMarker';

export type EmojiMarkerProps = {
  onClick: () => void;
};
export default EmojiMarker;
