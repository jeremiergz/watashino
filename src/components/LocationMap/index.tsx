import { GoogleMap, Marker, useLoadScript } from '@react-google-maps/api';
import { graphql, useStaticQuery } from 'gatsby';
import React from 'react';
import styled from 'styled-components';

const LocationMap = styled.div`
    height: 416px;
    width: 100%;
`;

const LocationMapComponent = () => {
    const { isLoaded } = useLoadScript({
        googleMapsApiKey: process.env.GOOGLE_MAPS_API_KEY,
        preventGoogleFontsLoading: true,
    });
    const {
        dataJson: { location },
    } = useStaticQuery<GraphQL.LocationMapDataQueryQuery>(graphql`
        query LocationMapDataQuery {
            dataJson {
                location {
                    country
                    city
                    lat
                    lng
                }
            }
        }
    `);
    const position = {
        lat: location.lat,
        lng: location.lng,
    };
    return (
        isLoaded && (
            <LocationMap>
                <GoogleMap
                    center={position}
                    mapContainerStyle={{ height: '100%', width: '100%', borderRadius: '8px' }}
                    zoom={4}
                >
                    <Marker position={position} />
                </GoogleMap>
            </LocationMap>
        )
    );
};

LocationMapComponent.displayName = 'LocationMap';

export default LocationMapComponent;
