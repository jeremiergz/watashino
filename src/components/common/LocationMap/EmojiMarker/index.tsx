import React, { useState } from 'react';
import EmojiAstonished from '../../../svg/EmojiAstonished';
import EmojiNerd from '../../../svg/EmojiNerd';
import EmojiUgly from '../../../svg/EmojiUgly';

type EmojiMarkerProps = {
  onClick: () => void;
};

const Emojis = {
  EmojiAstonished,
  EmojiNerd,
  EmojiUgly,
};

let timeout: number;

const EmojiMarker = ({ onClick }: EmojiMarkerProps) => {
  const [marker, setMarker] = useState(EmojiNerd.displayName);
  const handleMarkerClick = () => {
    if (typeof onClick === 'function') onClick();
    if (marker === EmojiNerd.displayName) {
      setMarker(EmojiAstonished.displayName);
    } else if (marker === EmojiAstonished.displayName) {
      setMarker(EmojiUgly.displayName);
    }
    clearTimeout(timeout);
    timeout = setTimeout(() => {
      setMarker(EmojiNerd.displayName);
    }, 2500);
  };
  const Emoji = Emojis[marker];
  return <Emoji cursor="help" onClick={handleMarkerClick} />;
};

EmojiMarker.displayName = 'EmojiMarker';

export default EmojiMarker;
