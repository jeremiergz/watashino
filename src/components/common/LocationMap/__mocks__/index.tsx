import React from 'react';

const LocationMap = () => {
  return <></>;
};

LocationMap.displayName = 'LocationMap';

export default LocationMap;
