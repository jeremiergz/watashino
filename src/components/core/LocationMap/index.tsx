import { GoogleMap, Marker, useLoadScript } from '@react-google-maps/api';
import { graphql, useStaticQuery } from 'gatsby';
import React from 'react';
import useTheme from '../../../hooks/useTheme';
import Box from '../../primitives/Box';

const LocationMap = () => {
  const theme = useTheme();
  const { isLoaded } = useLoadScript({
    googleMapsApiKey: process.env.GATSBY_GOOGLE_MAPS_API_KEY,
    preventGoogleFontsLoading: true,
  });
  const {
    dataJson: { location },
    googleMapsJson: { styles },
  } = useStaticQuery<GraphQL.LocationMapDataQuery>(graphql`
    query LocationMapData {
      dataJson {
        location {
          lat
          lng
        }
      }
      googleMapsJson {
        styles {
          dark {
            elementType
            featureType
            stylers {
              color
            }
          }
          light {
            elementType
            featureType
            stylers {
              color
            }
          }
        }
      }
    }
  `);
  const position = {
    lat: location.lat,
    lng: location.lng,
  };
  return (
    isLoaded && (
      <Box height={416} width="100%">
        <GoogleMap
          center={position}
          mapContainerStyle={{ height: '100%', width: '100%', borderRadius: '8px' }}
          options={{
            styles: [
              ...styles[theme.type],
              {
                elementType: 'geometry',
                featureType: 'water',
                stylers: [{ color: theme.type === 'light' ? theme.colors.primary : theme.colors.secondary }],
              },
            ],
          }}
          zoom={4}
        >
          <Marker position={position} />
        </GoogleMap>
      </Box>
    )
  );
};

LocationMap.displayName = 'LocationMap';

export default LocationMap;
