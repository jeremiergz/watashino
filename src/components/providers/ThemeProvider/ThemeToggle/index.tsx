import Button from 'components/Button';
import FlexBox from 'components/primitives/FlexBox';
import Text from 'components/primitives/Text';
import MoonIcon from 'components/svg/Moon';
import SunIcon from 'components/svg/Sun';
import React from 'react';
import { useTheming } from '..';

const iconMapping = {
  dark: MoonIcon,
  light: SunIcon,
};

const ThemeToggle: React.FC = () => {
  const { theme, toggle } = useTheming();
  const ThemeIcon = iconMapping[theme.type];
  return (
    <Button onClick={() => toggle()} position="absolute" right={3} top={3} variant="outlined" width={72}>
      <FlexBox alignItems="center" justifyContent="space-between" width="100%">
        <ThemeIcon height={20} width={20} />
        <Text color="gray" fontFamily={theme.fonts.main} fontSize={14}>
          {theme.type}
        </Text>
      </FlexBox>
    </Button>
  );
};

ThemeToggle.displayName = 'ThemeToggle';

export default ThemeToggle;
