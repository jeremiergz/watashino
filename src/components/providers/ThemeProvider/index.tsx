import { graphql, useStaticQuery } from 'gatsby';
import React, { createContext, useContext, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle, ThemeProvider as StyledThemeProvider } from 'styled-components';
import { Theme as StyledTheme } from 'styled-system';

const GlobalStyle = createGlobalStyle`
    html, body {
      background-color: ${({ theme }: { theme: Theme }) => theme.background};
      height: 100%;
      width: 100%;
      margin: 0;
      padding: 0;
      transition: background-color 250ms ease;
    }
    body {
      color: ${({ theme }: { theme: Theme }) => theme.colors.text};
      font-family: ${({ theme }: { theme: Theme }) => theme.fonts.main};
      font-weight: ${({ theme }: { theme: Theme }) => theme.fontWeights.regular};
    }
    #___gatsby {
      position: relative;
      min-height: 100%;
    }
    /* Format Markdown images loaded by gatsby-remark-images */
    .gatsby-resp-image-wrapper {
      .gatsby-resp-image-background-image, .gatsby-resp-image-image {
        border-radius: 8px;
      }
    }
    .grvsc-container {
      text-align: left;
    }
`;

const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'];
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const defaultMode = 'light';
const isBrowser = typeof window !== 'undefined';
const localStorageKey = 'theme:mode';

const ThemeContext = createContext<{ theme: Theme; toggle: () => void }>({ theme: null, toggle: () => null });

const ThemeProvider: React.FC = ({ children }) => {
  const { themeJson } = useStaticQuery<GraphQL.ThemeDataQuery>(graphql`
    query ThemeData {
      themeJson {
        colors {
          black
          dark
          error
          gray
          info
          lightgray
          primary
          secondary
          success
          vsCodeDark
          white
        }
        fontWeights {
          bold
          regular
          semiBold
        }
        fonts {
          main
        }
      }
    }
  `);
  const DarkTheme = useMemo(
    () => ({
      background: themeJson.colors.black,
      breakpoints,
      colors: {
        ...themeJson.colors,
        text: themeJson.colors.white,
      },
      fonts: themeJson.fonts,
      fontWeights: themeJson.fontWeights,
      svg: themeJson.colors.white,
      type: 'dark' as const,
    }),
    [themeJson],
  );
  const LightTheme = useMemo(
    () => ({
      background: themeJson.colors.white,
      breakpoints,
      colors: {
        ...themeJson.colors,
        text: '#262626',
      },
      fonts: themeJson.fonts,
      fontWeights: themeJson.fontWeights,
      svg: '#262626',
      type: 'light' as const,
    }),
    [themeJson],
  );
  const [isInitialized, setIsInitialized] = useState(false);
  const [mode, setMode] = useState<Theme['type']>(defaultMode);
  const toggle = () => {
    const newMode = mode === 'light' ? 'dark' : 'light';
    setMode(newMode);
  };
  const theme = mode === 'light' ? LightTheme : DarkTheme;
  useEffect(() => {
    if (isBrowser && isInitialized) localStorage.setItem(localStorageKey, mode);
  }, [mode]);
  useEffect(() => {
    const storedMode = (isBrowser && (localStorage.getItem(localStorageKey) as Theme['type'])) || defaultMode;
    setMode(storedMode);
    setIsInitialized(true);
  }, []);
  return (
    <ThemeContext.Provider value={{ theme, toggle }}>
      <StyledThemeProvider theme={theme}>
        <GlobalStyle />
        {children}
      </StyledThemeProvider>
    </ThemeContext.Provider>
  );
};

ThemeProvider.displayName = 'ThemeProvider';

/**
 * Returns the theming context that allows retrieval of theme & toggling theme mode.
 */
function useTheming() {
  return useContext(ThemeContext);
}

export { useTheming };
export type Theme = Omit<StyledTheme, 'breakpoints' | 'colors' | 'fonts' | 'fontWeights'> & {
  background: string;
  breakpoints: string[];
  colors: GraphQL.ThemeDataQuery['themeJson']['colors'] & { text?: string };
  fonts: GraphQL.ThemeDataQuery['themeJson']['fonts'];
  fontWeights: GraphQL.ThemeDataQuery['themeJson']['fontWeights'];
  type: 'dark' | 'light';
};
export type ThemeProps = {
  background?: string;
  colors?: keyof Theme['colors'] | 'inherit';
  fontWeight?: keyof Theme['fontWeights'];
  svg?: string;
  type?: Theme['type'];
};
export default ThemeProvider;
