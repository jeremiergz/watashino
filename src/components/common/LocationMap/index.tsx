import { graphql, useStaticQuery } from 'gatsby';
import Img from 'gatsby-image';
import { trackCustomEvent } from 'gatsby-plugin-google-analytics';
import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import { useTheming } from '../../core/ThemingManager';
import Box from '../../primitives/Box';
import EmojiMarker from './EmojiMarker';
import Link from '../Link';

const StaticGMapImg = styled(Img)`
  max-height: 416px;
  width: 100% !important;
  border-radius: 8px;
  img {
    object-fit: none !important;
  }
  @media (min-width: 641px) {
    img {
      object-fit: cover !important;
    }
  }
`;

const LocationMap: FunctionComponent = () => {
  const { mode } = useTheming();
  const handleMarkerClick = () => {
    trackCustomEvent({
      action: 'click',
      category: 'Google Maps Emoji Marker',
    });
  };
  const {
    gmapsImgs: { nodes: gmapsImgNodes },
  } = useStaticQuery(graphql`
    query LocationMapData {
      gmapsImgs: allStaticGmapsImage {
        nodes {
          childFile {
            childImageSharp {
              fixed(height: 640) {
                ...GatsbyImageSharpFixed
              }
            }
          }
          mapURL
          theme
        }
      }
    }
  `);
  const gmapsImgNode = gmapsImgNodes.find(i => i.theme === mode) || gmapsImgNodes.find(i => i.theme === 'default');
  const locationMap = gmapsImgNode.childFile.childImageSharp.fixed;
  return (
    <Box position="relative">
      <Link external to={gmapsImgNode.mapURL}>
        <StaticGMapImg alt="Location" fixed={locationMap} />
      </Link>
      <Box bottom={0} height={32} left={0} margin="auto" position="absolute" right={0} top={0} width={32}>
        <EmojiMarker onClick={handleMarkerClick} />
      </Box>
    </Box>
  );
};

LocationMap.displayName = 'LocationMap';

export default LocationMap;
