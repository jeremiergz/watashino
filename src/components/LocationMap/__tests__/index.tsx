import ThemeProvider from 'components/providers/ThemeProvider';
import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import LocationMap from '../../LocationMap';

describe('components/LocationMap Test Suite', () => {
  beforeAll(() => {
    const childImageSharp = {
      fixed: {
        aspectRatio: 1,
        base64: 'data:image/png;base64',
        src: '/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png',
        srcSet:
          '/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,\n/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,\n/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x',
      },
    };
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => ({
      gmapsImgs: { nodes: [{ mapURL: 'https://fake.com', theme: 'default', childFile: { childImageSharp } }] },
      themeJson: {
        colors: {
          black: '#212121',
          dark: '#424242',
          error: '#f44336',
          gray: '#757575',
          info: '#2196f3',
          lightgray: '#e0e0e0',
          primary: '#31859a',
          secondary: '#17365c',
          success: '#4caf50',
          vsCodeDark: '#1e1e1e',
          white: '#fafafa',
        },
        fonts: {
          main: "'Open Sans', sans-serif",
        },
        fontWeights: {
          bold: 800,
          regular: 400,
          semiBold: 600,
        },
      },
    }));
  });

  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider>
        <LocationMap />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
