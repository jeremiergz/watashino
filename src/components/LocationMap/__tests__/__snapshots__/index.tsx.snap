// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LocationMap Test Suite matches snapshot 1`] = `
<div
  className="sc-AxirZ kdLUNZ"
>
  <a
    className="sc-AxirZ sc-AxiKw gKjgFD"
    href="https://fake.com"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      className="sc-AxjAm sc-AxhUy bEWhpf gatsby-image-wrapper"
      style={
        Object {
          "display": "inline-block",
          "height": undefined,
          "overflow": "hidden",
          "position": "relative",
          "width": undefined,
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/png;base64"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          srcSet="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x"
        />
        <img
          alt="Location"
          loading="lazy"
          onLoad={[Function]}
          src="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png"
          srcSet="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x\\" /><img loading=\\"lazy\\" srcset=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x\\" src=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png\\" alt=\\"Location\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
  <div
    className="sc-AxirZ lfbEcI"
    height={32}
    width={32}
  >
    <svg
      className="sc-AxhCb jeQFck"
      cursor="help"
      onClick={[Function]}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="256"
        cy="256"
        fill="#ffd93b"
        r="256"
      />
      <path
        d="M351.936 367.216c-7.36 40.8-39.84 81.92-96 81.92s-88.64-41.12-95.84-81.92c-1.12-6.08 3.04-11.84 9.12-12.96 6.08-.96 11.84 3.04 12.96 9.12 5.6 31.52 30.56 63.36 73.76 63.36s68.32-31.84 73.92-63.36c1.12-6.08 6.88-10.08 12.96-9.12 6.08 1.104 10.24 6.88 9.12 12.96z"
        fill="#3e4347"
      />
      <path
        d="M512 256c0 141.44-114.64 256-256 256-80.48 0-152.32-37.12-199.28-95.28 43.92 35.52 99.84 56.72 160.72 56.72 141.36 0 256-114.56 256-256 0-60.88-21.2-116.8-56.72-160.72C474.8 103.68 512 175.52 512 256z"
        fill="#f4c534"
      />
      <path
        d="M235.44 225.232c0 43.808-35.424 79.376-79.376 79.376-43.792 0-79.376-35.568-79.376-79.376 0-43.952 35.584-79.52 79.376-79.52 43.952 0 79.376 35.584 79.376 79.52z"
        fill="#fff"
      />
      <circle
        cx="156.064"
        cy="225.12"
        fill="#3e4347"
        r="36.16"
      />
      <path
        d="M276.624 225.088c0-43.808 35.424-79.376 79.376-79.376 43.808 0 79.376 35.568 79.376 79.376 0 43.952-35.568 79.52-79.376 79.52-43.952.016-79.376-35.568-79.376-79.52z"
        fill="#fff"
      />
      <path
        d="M485.264 213.168H445.76c-5.92-44.336-43.888-78.64-89.76-78.64-33.952 0-63.552 18.816-79.04 46.56-4.208 7.536-12.16 11.92-20.784 11.872h-.384c-8.592.064-16.512-4.336-20.688-11.84-15.488-27.76-45.088-46.592-79.04-46.592-45.856 0-83.744 34.336-89.68 78.64H26.736v24h39.92c5.568 41.408 39.216 74.736 81.6 78.32 53.632 4.544 99.44-38.224 98.304-92.144.912-4.512 4.72-7.968 9.472-7.968 4.864 0 9.376 3.504 9.376 9.872 0 52.288 44.56 94.384 97.696 90.304 42.528-3.264 76.832-36.528 82.368-78.384h39.776v-24h.016zm-329.2 80.224c-37.584 0-68.176-30.592-68.176-68.16 0-37.664 30.576-68.304 68.176-68.304 37.584 0 68.16 30.64 68.16 68.304 0 37.584-30.576 68.16-68.16 68.16zm199.936 0c-37.584 0-68.176-30.592-68.176-68.16 0-37.664 30.576-68.304 68.176-68.304 37.584 0 68.16 30.64 68.16 68.304 0 37.584-30.576 68.16-68.16 68.16z"
        fill="#3e4347"
      />
      <ellipse
        cx="169.695"
        cy="211.503"
        fill="#fff"
        rx="10"
        ry="6.48"
        transform="rotate(-135 169.696 211.505)"
      />
      <circle
        cx="356"
        cy="225.12"
        fill="#3e4347"
        r="36.16"
      />
      <ellipse
        cx="369.623"
        cy="211.559"
        fill="#fff"
        rx="10"
        ry="6.48"
        transform="rotate(-135 369.625 211.56)"
      />
      <path
        d="M413.776 107.456c-5.12 2.992-12.256 1.536-15.52-4.064-26.672-45.648-82.272-35.552-82.864-35.408-6.176 1.2-12.048-2.864-13.264-9.04-1.056-6.176 2.864-12.064 9.04-13.264 2.864-.448 71.872-12.96 106.672 46.256 3.168 5.424 1.36 12.368-4.064 15.52zm-315.552 0c5.12 2.992 12.256 1.536 15.52-4.064C140.416 57.744 196 67.84 196.608 67.984c6.176 1.2 12.048-2.864 13.264-9.04 1.056-6.176-2.864-12.064-9.04-13.264-2.864-.448-71.872-12.96-106.672 46.256-3.168 5.424-1.36 12.368 4.064 15.52z"
        fill="#3e4347"
      />
    </svg>
  </div>
</div>
`;
